src/native/libs/System.Native/pal_runtimeinformation.c

```diff
@@ -48,8 +53,93 @@ int32_t SystemNative_GetUnixVersion(char* version, int* capacity)
     return 0;
 }
 
-/* Returns an int representing the OS Architecture. -1 if same as process architecture. */
+// Keep in sync with System.Runtime.InteropServices.Architecture enum
+enum
+{
+    ARCH_X86,
+    ARCH_X64,
+    ARCH_ARM,
+    ARCH_ARM64,
+    ARCH_WASM,
+    ARCH_S390X,
+    ARCH_LOONGARCH64,
+    ARCH_ARMV6,
+};
+
 int32_t SystemNative_GetOSArchitecture()
 {
-    return -1;
+#ifdef TARGET_WASM
+    return ARCH_WASM;
+#else
+    int32_t result = -1;
+#ifdef TARGET_SUNOS
+    // On illumos/Solaris, the recommended way to obtain machine
+    // architecture is using `sysinfo` rather than `utsname.machine`.
+
+    char isa[32];
+    if (sysinfo(SI_ARCHITECTURE_K, isa, sizeof(isa)) > -1)
+    {
+        {
+#else
+    {
+        struct utsname _utsname;
+        if (uname(&_utsname) > -1)
+        {
+            char* isa = _utsname.machine;
+#endif
```

Comment:
```suggestion
    char isa[32];
    if (sysinfo(SI_ARCHITECTURE_K, isa, sizeof(isa)) > -1)
    {
#else
    struct utsname _utsname;
    if (uname(&_utsname) > -1)
    {
        char* isa = _utsname.machine;
#endif
```
Are both scopes necessary here?

Reply:
Nope, I just added them because indentation was looking funny. :)

Reply:
The delta that I have suggested does not look funny to me. Am I missing something?

Reply:
Ah, I thought I already pushed this one from mac (I am using three machines for this PR). 😅
we need to erase a closing brace too, let me fix it.

Reply:
Pushed.